bindings/core/src/method_handler/utils.rs

```
@@ -129,7 +129,11 @@ pub(crate) fn call_utils_method_internal(method: UtilsMethod) -> Result<Response
                 protocol_parameters,
             );
 
-            Response::TransactionFailureReason(context.validate()?)
+            if let Err(e) = context.validate() {
+                Response::TransactionFailureReason(Some(e))
+            } else {
+                Response::TransactionFailureReason(None)
+            }
```

Don't we now want to return an error in the bindings as well?

---

So removing ` Response::TransactionFailureReason` and letting `validate` error?

---

Yes, but not sure if that's really better, that's why I made it a question

---

Now that it's an actual error, maybe it should behave like one and not return an option yeah

---

I think it's slightly better actually

---

Done